### PR TITLE
IGNITE-14476 Get rid of using storage implementation explicitly in ConfigurationRoot annotation

### PR DESCRIPTION
**Pull request by SammyVimes**

### CODE REVIEW

File: modules/configuration/src/main/java/org/apache/ignite/configuration/storage/ConfigurationStorageType.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration.storage;
+
+/**
+ * Type of configuration storage.
+ */
+public enum ConfigurationStorageType {
```
Comment (alievmirza) [inline modules/configuration/src/main/java/org/apache/ignite/configuration/storage/ConfigurationStorageType.java:23]: Seems that there should be two types of configurations storage: `LOCAL` and `DISTRIBUTED`, for Vault and Metastorage correspondingly

File: modules/configuration-annotation-processor/src/test/java/org/apache/ignite/configuration/notifications/ConfigurationListenerTest.java [hunk 2]
```diff
@@ -70,9 +71,11 @@
     public void before() {
         registry.registerRootKey(ParentConfiguration.KEY);
 
-        registry.registerStorage(new TestConfigurationStorage());
+        TestConfigurationStorage storage = new TestConfigurationStorage();
 
-        registry.startStorageConfigurations(TestConfigurationStorage.class);
+        registry.registerStorage(storage);
```
Comment (sashapolo) [inline modules/configuration-annotation-processor/src/test/java/org/apache/ignite/configuration/notifications/ConfigurationListenerTest.java:76]: why do `registerStorage` and `startStorageConfigurations` have to be two separate methods?

Comment (SammyVimes) [reply modules/configuration-annotation-processor/src/test/java/org/apache/ignite/configuration/notifications/ConfigurationListenerTest.java:76]: I can't really say, lets wait for @ibessonov

**Review by sashapolo — APPROVED** [review]

File: modules/configuration-annotation-processor/src/main/java/org/apache/ignite/configuration/processor/internal/Processor.java [hunk 3]
```diff
@@ -96,6 +98,18 @@ public Processor() {
...
+        } catch (Throwable t) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            t.printStackTrace(pw);
+            processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, "Failed to process configuration: " + sw.toString());
+        }
+        return false;
+    }
+
+    private boolean process0(RoundEnvironment roundEnvironment) {
```
Comment (sergey-chugunov-1985) [inline modules/configuration-annotation-processor/src/main/java/org/apache/ignite/configuration/processor/internal/Processor.java:112]: Method without comment

**Review by sergey-chugunov-1985 — CHANGES_REQUESTED** [review]
I left minor comment on the PR, otherwise patch looks good to me.

**Comment by ibessonov** [comment]
Merged manually.

**Pull request closed without merging**